src/openrct2/scripting/bindings/game/ScContext.hpp

```diff
@@ -45,6 +45,36 @@ namespace OpenRCT2::Scripting
         }
 
     private:
+        std::vector<DukValue> allPlugins_get()
+        {
+            // Get all of the plugins from the script engine
+            OpenRCT2::Scripting::ScriptEngine& scriptEngine = GetContext()->GetScriptEngine();
+            std::vector<std::shared_ptr<OpenRCT2::Scripting::Plugin>> allPlugins = scriptEngine.GetPlugins();
+            auto ctx = scriptEngine.GetContext();
+            std::vector<DukValue> result;
+            // Iterate through all plugins and and cast their data to Duk objects
+            for (std::shared_ptr<OpenRCT2::Scripting::Plugin> plugin_ptr : allPlugins)
+            {
+                // Pull out metadata
+                OpenRCT2::Scripting::Plugin& plugin = *plugin_ptr;
+                OpenRCT2::Scripting::PluginMetadata metadata = plugin.GetMetadata();
+                DukObject dObj(ctx);
+                // Set properties of Duk object
+                dObj.Set("name", metadata.Name);
+                dObj.Set("version", metadata.Version);
+                // Concatenate all authors into one string
+                std::string authors = "";
+                for (std::string s : metadata.Authors)
+                {
+                    authors += s + " ";
+                }
+
+                dObj.Set("authors", authors);
```

Comment:
Would be better to return an array of authors, plugins can decide how to deal with displaying this data. In javascript this is quite easy with a join call

Reply:
Definitely is better to return the array of authors instead, just took some playing around to get it to work. Before, I was using the DukObj class to encapsulate the PluginMetadata, but that didn't have a Set for string arrays. I'm now just interacting directly with the Duk stack to cast the metadata now.